qa/L0_java_memory_growth/MemoryGrowthTest.java

```diff
@@ -378,26 +386,36 @@ output0, output1, new FloatPointer(input0_data), new FloatPointer(input1_data),
 
       // Measure memory growth without outliers by taking difference
       // between 90th percentile and 10th percentile memory usage.
+      final double bytes_in_mb = 1E6;
       Collections.sort(memory_snapshots);
-      int index_90 = ((int) Math.ceil(90.0 / 100.0 * memory_snapshots.size())) - 1;
-      int index_10 = ((int) Math.ceil(10.0 / 100.0 * memory_snapshots.size())) - 1;
-      double memory_allocation_delta = memory_snapshots.get(index_90) - memory_snapshots.get(index_10);
-      double memory_allocation_delta_mb = memory_allocation_delta / 1E6;
-      double memory_allocation_delta_percent = memory_allocation_delta / memory_snapshots.get(index_90);
+      int index_max = ((int) Math.ceil(max_percentile / 100.0 * memory_snapshots.size())) - 1;
+      int index_min = ((int) Math.ceil(min_percentile / 100.0 * memory_snapshots.size())) - 1;
+      double memory_allocation_delta = memory_snapshots.get(index_max) - memory_snapshots.get(index_min);
+      double memory_allocation_delta_mb = memory_allocation_delta / bytes_in_mb;
+      double memory_allocation_delta_percent = memory_allocation_delta / memory_snapshots.get(index_max);
 
       System.out.println("Change in memory allocation (MB): " +
           memory_allocation_delta_mb + ", " +
-          memory_allocation_delta_percent + "%");
+          (memory_allocation_delta_percent * 100) + "%");
 
-      if(memory_allocation_delta_percent <= max_allowed){
-        return true;
-      } else {
-        return false;
+      boolean passed = true;
+
+      if(memory_allocation_delta_percent >= max_growth_allowed){
+        passed = false;
+        System.out.println("Exceeded allowed memory growth (" +
+          (max_growth_allowed * 100) + "%)");
       }
+
+      if((memory_snapshots.get(index_max) / bytes_in_mb) >= max_mem_allowed){
+        passed = false;
+        System.out.println("Exceeded allowed memory (" + max_mem_allowed + 
+          "MB), got " + memory_allocation_delta_mb + "MB");
+      }
+      return passed;
```

Comment:
Just to confirm, so test passes only if `(< %) && (< delta)`? And should we further limit the delta allowed? From the test it seems to be much less than 30 MB even for long running setting

Reply:
The test only passes if the delta is (< %) and the maximum memory used is (<max_mem_allowed). We can reduce it further to 20MB. (It usually runs in the 15-17MB range). The 30MB was to provide more slack to prevent intermittence from Java's garbage collection.

Reply:
I see, the limit is on total memory usage.